### ISSUE TITLE
Error in hash of Rational{Int}

### ISSUE THREAD

**Issue by Jutho** [JuliaLang/julia#25814]
There is a subtle bug in the hashing code of `Rational{<:BigInteger64}`, in the file hashing2.jl, line 158:
```julia
            return hash(ldexp(Float64(num),pow))
```
This does not pass the incoming hash argument `h`, and thus, while it produces the correct result when just comparing `hash(3//2)` with `hash(1.5)`, it does not produce the same result for `hash(3//2,h)` and `hash(1.5,h)` for any non-zero `h`.

I would fix myself but I am traveling and hence not in my normal workflow.

Pinging @StefanKarpinski as I think he wrote the hashing code.

**Comment by StefanKarpinski** [comment]
Good catch! Yes, this seems like a simple enough fix – just pass the `h` argument on this line:

https://github.com/JuliaLang/julia/blob/ac389b016702666853bdb80d521a02977705bba1/base/hashing2.jl#L158

Then add a test to verify that this is done correctly.

**Comment by StefanKarpinski** [comment]
Note that we're not going to backport this to 0.4 or 0.5 at this point, but I've tagged this anyway since it *could* be backported to those releases.

**Comment by KristofferC** [comment]
What prevents it from being backported to 0.3 though?

**Comment by StefanKarpinski** [comment]
Only the lack of a label. I drew the line at creating a label for backports that we will never do.
